### ISSUE TITLE
BlockSigningRoot is redundant

### ISSUE THREAD

**Issue by terencechain** [prysmaticlabs/prysm#2937]
[BlockSigningRoot](https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/shared/blockutil/beacon_block.go#L11) feels redundant. The worse part is it requires a tree hash and signing root, why not just use `ssz.SigningRoot(beaconBlock)`

https://github.com/prysmaticlabs/prysm/blob/35de6cc816023537d91d009f507b5ab7bf096d5a/shared/blockutil/beacon_block.go#L8-L24

**Comment by prestonvanloon** [comment]
I think I agree that this is not correct.

```python
def process_block_header(state: BeaconState, block: BeaconBlock) -> None:
    # Verify that the slots match
    assert block.slot == state.slot
    # Verify that the parent matches
    assert block.parent_root == signing_root(state.latest_block_header)
    # Save current block as the new latest block
    state.latest_block_header = BeaconBlockHeader(
        slot=block.slot,
        parent_root=block.parent_root,
        state_root=Hash(),  # Overwritten in the next `process_slot` call
        body_root=hash_tree_root(block.body),
    )
    # Verify proposer is not slashed
    proposer = state.validators[get_beacon_proposer_index(state)]
    assert not proposer.slashed
    # Verify proposer signature
    assert bls_verify(proposer.pubkey, signing_root(block), block.signature, get_domain(state, DOMAIN_BEACON_PROPOSER))
```

https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#block-header

When we verify the signature, we are taking the signature of the entire block.

It's a bit confusing because the parent root is the header, but the root signed by the validator is the entire block, not just the header.

**Comment by nisdas** [comment]
Closed by #2945
